feat(minifier): complete implementation of statement fusion

camc314 opened the pull request:
- enables all test in statement fusion
 - implements some more node in `may_have_side_effects`

camc314 commented:
* **#6566** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6566?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camc314 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented on crates/oxc_minifier/src/ast_passes/statement_fusion.rs:
```diff
@@ -74,7 +74,7 @@ impl<'a> StatementFusion {
                     || for_stmt.init.as_ref().is_some_and(ForStatementInit::is_expression)
             }
             // TODO: support for-in, we need to check the init for side effects
```
```suggestion
```

Boshen commented:
### Merge activity

* **Oct 14, 10:08 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
feat(minifier): complete implementation of statement fusion (#6566)

- enables all test in statement fusion
 - implements some more node in `may_have_side_effects`
